pkg/api/frontend_logging_test.go

```diff
@@ -0,0 +1,149 @@
+package api
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	"github.com/grafana/grafana/pkg/models"
+	log "github.com/inconshreveable/log15"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type logScenarioFunc func(c *scenarioContext, logs []*log.Record)
+
+func logSentryEventScenario(t *testing.T, desc string, event frontendSentryEvent, fn logScenarioFunc) {
+	t.Run(desc, func(t *testing.T) {
+		logs := []*log.Record{}
+		origHandler := frontendLogger.GetHandler()
+		frontendLogger.SetHandler(log.FuncHandler(func(r *log.Record) error {
+			logs = append(logs, r)
+			return nil
+		}))
+		t.Cleanup(func() {
+			frontendLogger.SetHandler(origHandler)
+		})
+
+		sc := setupScenarioContext("/log")
+		hs := HTTPServer{}
+
+		sc.defaultHandler = Wrap(func(w http.ResponseWriter, c *models.ReqContext) Response {
+			sc.context = c
+			return hs.logFrontendMessage(c, event)
+		})
+
+		sc.m.Post(sc.url, sc.defaultHandler)
+		sc.fakeReqWithParams("POST", sc.url, map[string]string{}).exec()
+		fn(sc, logs)
+	})
+}
+
+func TestFrontendLoggingEndpoint(t *testing.T) {
+	ts, err := time.Parse("2006-01-02T15:04:05.000Z", "2020-10-22T06:29:29.078Z")
+	require.NoError(t, err)
+
+	t.Run("FrontendLoggingEndpoint", func(t *testing.T) {
+		request := sentry.Request{
+			URL: "http://localhost:3000/",
+			Headers: map[string]string{
+				"User-Agent": "Chrome",
+			},
+		}
+
+		user := sentry.User{
+			Email: "geralt@kaermorhen.com",
+			ID:    "45",
+		}
+
+		errorEvent := frontendSentryEvent{
+			&sentry.Event{
+				EventID:   "123",
+				Level:     sentry.LevelError,
+				Request:   &request,
+				Timestamp: ts,
+			},
+			&frontendSentryException{
+				Values: []frontendSentryExceptionValue{
+					{
+						Type:  "UserError",
+						Value: "Please replace user and try again",
+						Stacktrace: sentry.Stacktrace{
+							Frames: []sentry.Frame{
+								{
+									Function: "foofn",
+									Filename: "foo.js",
+									Lineno:   123,
+									Colno:    23,
+								},
+								{
+									Function: "barfn",
+									Filename: "bar.js",
+									Lineno:   113,
+									Colno:    231,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		logSentryEventScenario(t, "Should log received error event", errorEvent, func(sc *scenarioContext, logs []*log.Record) {
+			assert.Equal(t, sc.resp.Code, 200)
```

Comment:
With assert.Equal, the expectation comes before the actual value.
```suggestion
			assert.Equal(t, 200, sc.resp.Code)
```